sklearn/model_selection/_search.py

```diff
@@ -788,9 +834,29 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
         # Store a list of param dicts at the key 'params'
         results['params'] = candidate_params
 
-        # NOTE test_sample counts (weights) remain the same for all candidates
-        test_sample_counts = np.array(test_sample_counts[:n_splits],
-                                      dtype=np.int)
+        # training train_sample_weight_sums needs to be done first because
+        # test_sample_counts overrides itself in the IF statement below with
+        # test_sample_weight_sums.
+        if self.return_train_score:
+            if train_sample_weight_sums is None:
+                samples = int(np.sum(test_sample_counts[:n_splits]))
+                train_sample_counts = \
+                    np.array(test_sample_counts[:n_splits], dtype=np.int) - \
```

Comment:
Could you explain why this difference is necessary?

Reply:
@zachary-mcpherson-zefr: Good catch!  The arguments are backward.  Will fix.  In the meantime, here's the reasoning (assuming the code was correct):

Say that you have the following three folds for cross validation:

* **fold 1**: *50* examples
* **fold 2**: *51* examples
* **fold 3**: *49* examples

So it typically works by iterating over the test fold.  So let's say that we are on the second loop iteration and therefore, **fold 2** is the test fold.  That means that folds 1 and 3 are the training folds.  So, there should be *99* training examples for test **fold 2**.

`samples` should represent the total number of rows of data in the entire dataset.  Then, to get the number of training rows, we can subtract the number of testing rows from the total to get the number of training rows associated with the test fold.  So, back to our example, if the test fold is **fold 2**, we have `samples = 150` and **fold 2** is *51*, then `150 - 51 == 99`.  Make sense?

Reply:
Yeah that makes sense, thanks!